[Nightly test failure] rclpy.test_events_ececutor

### Generated by Generative AI

_No response_

### Operating System:

Ubuntu 24.04

### ROS version or commit hash:

rolling, kilted

### RMW implementation (if applicable):

rmw_zenoh_cpp

### RMW Configuration (if applicable):

_No response_

### Client library (if applicable):

_No response_

### 'ros2 doctor --report' output

<details><summary>ros2 doctor --report</summary>

```console
<COPY OUTPUT HERE>
```
</details>


### Steps to reproduce issue

https://build.ros2.org/job/Rci__nightly-zenoh_ubuntu_noble_amd64/228/testReport/rclpy.rclpy.test.test_events_executor/TestEventsExecutor/test_pub_sub/

### Expected behavior

Test passes

### Actual behavior

```
self = <test.test_events_executor.TestEventsExecutor testMethod=runTest>

    def test_pub_sub(self) -> None:
        sub_node = SubTestNode()
        new_pub_future = sub_node.expect_pub_info()
        received_future = sub_node.expect_message()
        self.executor.add_node(sub_node)
    
        # With subscriber node alone, should be no publisher or messages
>       self._expect_future_not_done(new_pub_future)

../../src/ros2/rclpy/rclpy/test/test_events_executor.py:525: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../src/ros2/rclpy/rclpy/test/test_events_executor.py:465: in _expect_future_not_done
    self.assertFalse(future.done())
E   AssertionError: True is not false
```

### Additional information

_No response_

--- a/rclpy/test/test_events_executor.py
+++ b/rclpy/test/test_events_executor.py
@@ -448,6 +448,11 @@ class TestEventsExecutor(unittest.TestCase):
         self.executor = rclpy.experimental.EventsExecutor()
 
     def tearDown(self) -> None:
+        # Clean up all nodes still in the executor before shutdown
+        for node in self.executor.get_nodes():
+            self.executor.remove_node(node)
+            node.destroy_node()
+        self.executor.shutdown()
         rclpy.shutdown()
 
     def _expect_future_done(self, future: rclpy.Future[typing.Any]) -> None:

